### DISCUSSION TITLE
LightGlue Params

### DISCUSSION THREAD

**Discussion by DiZhang980** [fabio-sim/LightGlue-ONNX#48] [Q&A] [answered]
How to determine the parameter quantity of each layer for LightGlue? @fabio-sim

**Comment by fabio-sim** [comment, marked as answer]
Hi @DiZhang980, thank you for your interest in LightGlue-ONNX.

Assuming you mean each transformer layer, you can add this print statement:
```python
print(f"Layer {i+1}: {sum(p.numel() for p in self.transformers[i].parameters())} parameters")
```
to the following loop:
https://github.com/fabio-sim/LightGlue-ONNX/blob/e14a259a5d62fa058da2b9055d8268a982c25ccc/lightglue_onnx/lightglue.py#L328-L330

I get the following output:
```text
Layer 1: 1250560 parameters
Layer 2: 1250560 parameters
Layer 3: 1250560 parameters
Layer 4: 1250560 parameters
Layer 5: 1250560 parameters
Layer 6: 1250560 parameters
Layer 7: 1250560 parameters
Layer 8: 1250560 parameters
Layer 9: 1250560 parameters
```

**Reply by DiZhang980** [reply]
Thank you very much for your reply. I will try your statement.
